code/modules/mob/living/carbon/xenomorph/castes/warlock/abilities_warlock.dm

```diff
@@ -0,0 +1,409 @@
+			upper_right = locate(owner.x + 2, owner.y + 1, owner.z)
+
+	for(var/turf/affected_tile in block(lower_left, upper_right)) //everything in the 2x3 block is found.
+		affected_tile.Shake(4, 4, 2 SECONDS)
+		for(var/i in affected_tile)
+			var/atom/movable/affected = i
+			if(!ishuman(affected) && !istype(affected, /obj/item) && !isdroid(affected))
+				affected.Shake(4, 4, 20)
+				continue
+			if(ishuman(affected))
+				var/mob/living/carbon/human/H = affected
+				if(H.stat == DEAD)
+					continue
+				H.apply_effects(1, 1)
+				shake_camera(H, 2, 1)
+			var/throwlocation = affected.loc
+			for(var/x in 1 to 6)
+				throwlocation = get_step(throwlocation, owner.dir)
+			affected.throw_at(throwlocation, 4, 1, owner, TRUE)
+
+	owner.visible_message(span_xenowarning("[owner] sends out a huge blast of psychic energy!"), \
+	span_xenowarning("We send out a huge blast of psychic energy!"))
+
+	playsound(owner,'sound/effects/bamf.ogg', 75, TRUE)
+	playsound(owner, "alien_roar", 50)
+	//GLOB.round_statistics.psy_shield_blasts++
+	//SSblackbox.record_feedback("tally", "round_statistics", 1, "psy_shield_blasts")
+	succeed_activate()
+
+/obj/effect/xeno/shield
+	icon = 'icons/Xeno/96x96.dmi'
+	icon_state = "shield"
+	resistance_flags = RESIST_ALL
+	layer = ABOVE_MOB_LAYER
+	var/mob/living/carbon/xenomorph/owner
+
+/obj/effect/xeno/shield/Initialize(loc, creator)
+	. = ..()
+	owner = creator
+	dir = owner.dir
+	if(dir == EAST || dir == WEST)
+		bound_height = 96
+		bound_y = -32
+		pixel_y = -32
+	else
+		bound_width = 96
+		bound_x = -32
+		pixel_x = -32
+
+/obj/effect/xeno/shield/projectile_hit(obj/projectile/proj, cardinal_move, uncrossing)
+	if(!(cardinal_move & REVERSE_DIR(dir)))
+		if(!uncrossing)
+			proj.uncross_scheduled += src
+		return FALSE
+	if (uncrossing)
+		return FALSE
+	reflect_projectile(proj)
+	return FALSE
+
+///Reflects projectiles based on their relative incoming angle
+/obj/effect/xeno/shield/proc/reflect_projectile(obj/projectile/proj)
+	playsound(loc, 'sound/effects/portal.ogg', 20)
+	var/new_range = proj.proj_max_range - proj.distance_travelled
+
+	if(new_range <= 0)
+		proj.ammo.do_at_max_range(proj)
+		qdel(proj)
+		return
+	var/perpendicular_angle = Get_Angle(src, get_step(src, dir)) //the angle src is facing
+	var/new_angle = (perpendicular_angle + (perpendicular_angle - proj.dir_angle - 180))
+	if(new_angle < 0)
+		new_angle += 360
+	else if(new_angle > 360)
+		new_angle -= 360
+	proj.firer = src
+	proj.fire_at(shooter = src, source = src, range = new_range, angle = new_angle, recursivity = TRUE) //loc_override = loc //note investigate hitscan issue
+
+// ***************************************
+// *********** psychic crush
+// ***************************************
+/datum/action/xeno_action/activable/psy_crush
+	name = "psychic crush"
+	action_icon_state = "psy_crush"
+	mechanics_text = "Channel our psychic force to crush our enemies. The longer we channel, the larger the area."
```

Comment:
```suggestion
	mechanics_text = "Channel an expanding AOE crush effect, activating it again pre-maturely crushes enemies over an area. The more you stand still, the greater area it will affect, but you will cause more plasma."
```

Reply:
You can move while channeling crush